File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/models/FilmList.java

package com.dani_chuks.andeladeveloper.starwars.data.models;

import android.arch.persistence.room.Entity;

import com.dani_chuks.andeladeveloper.starwars.data.models.entities.Film;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

@Entity
public class FilmList {

    @SerializedName("count")
    @Expose
    private Integer count;
    @SerializedName("next")
    @Expose
    private Object next;
    @SerializedName("previous")
    @Expose
    private Object previous;
    @SerializedName("results")
    @Expose
    private List<Film> film = null;

    public Integer getCount() {
        return count;
    }

    public void setCount(Integer count) {
        this.count = count;
    }

    public Object getNext() {
        return next;
    }

    public void setNext(Object next) {
        this.next = next;
    }

    public Object getPrevious() {
        return previous;
    }

    public void setPrevious(Object previous) {
        this.previous = previous;
    }

    public List<Film> getFilm() {
        return film;
    }

    public void setFilm(List<Film> film) {
        this.film = film;
    }

}

File: app/src/main/java/com/dani_chuks/andeladeveloper/starwars/data/models/FilmList.kt
package com.dani_chuks.andeladeveloper.starwars.data.models

import androidx.room.Entity

import com.dani_chuks.andeladeveloper.starwars.data.models.entities.Film
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

@Entity
class FilmList {

    @SerializedName("count")
    @Expose
    var count: Int? = null
    @SerializedName("next")
    @Expose
    var next: Any? = null
    @SerializedName("previous")
    @Expose
    var previous: Any? = null
    @SerializedName("results")
    @Expose
    var film: List<Film>? = null

}
